internal/p2p/transaction_cache.go

```diff
@@ -0,0 +1,105 @@
+package p2p
+
+import (
+	"encoding/hex"
+	"sync"
+	"time"
+
+	log "github.com/koinos/koinos-log-golang"
+	"github.com/koinos/koinos-proto-golang/koinos/protocol"
+)
+
+type void struct{}
+
+// TransactionCacheItem is a an item in the transaction cache
+type TransactionCacheItem struct {
+	transactionID string
+	timeAdded     time.Time
+}
+
+// TransactionCache is a cache of recently received transactions
+type TransactionCache struct {
+	transactionMap   map[string]void
+	transactionItems []*TransactionCacheItem
+	cacheDuration    time.Duration
+	mu               sync.Mutex
+}
+
+// NewTransactionCache creates a new transaction cache
+func NewTransactionCache(cacheDuration time.Duration) *TransactionCache {
+	txc := &TransactionCache{
+		transactionMap:   make(map[string]void),
+		transactionItems: make([]*TransactionCacheItem, 0),
+		cacheDuration:    cacheDuration,
+	}
+
+	return txc
+}
+
+func (txc *TransactionCache) addTransactionItem(item *TransactionCacheItem) {
+	// Maintain the constraint that the transaction cache is sorted by time added
+	// Since this is not user facing code, we issue a panic, as this should never happen
+	numItems := len(txc.transactionItems)
+	older := numItems != 0 && item.timeAdded.Before(txc.transactionItems[numItems-1].timeAdded)
+	if older {
```

Comment:
You could refactor this since `older` isn't used after this check:

``` golang
if numItems != 0 && item.timeAdded.Before(txc.transactionItems[numItems-1].timeAdded) {
   panic("TransactionCache.addTransactionItem: transaction is older than the last transaction")
}
```